### PR TITLE
Support `if TYPE_CHECKING:` imports

### LINKED ISSUE #20: Allows sorting `if TYPE_CHECKING:` blocks
Currently zimports sorts only module level imports without indentation. Since when using type annotations importing some symbols in a guarded block is common, it would be nice if these imports could be sorted

### PR DESCRIPTION
**Pull request by CaselIT**
Added support for sorting type checking blocks
This is implemented by employing two passes, the fist only takes into account the type checking blocks, the second is the same pass as the current version

Fixes #20

This contains #21

### CODE REVIEW

File: zimports/zimports.py [hunk 1]
```diff
@@ -398,13 +498,15 @@ def _drill_for_warnings(options, filename, source_lines, warnings):
...
-            # we only deal with "top level" imports for now. imports
-            # inside of conditionals or in defs aren't counted.
-            whitespace = re.match(
-                r"^\s*", source_lines[warning.lineno - 1]
-            ).group(0)
-            if whitespace:
-                continue
+            line = source_lines[warning.lineno - 1]
+            if top_level:
+                # we only deal with "top level" imports for now. imports
```
Comment (zzzeek) [inline zimports/zimports.py]: this comment is obsolete :)

**Review by zzzeek — APPROVED** [review]
passes all tests and I just ran it on sqlalchemy source just to make sure, looks great!

**Comment by CaselIT** [comment]
For sqlalchemy I think we will need to add `-kt` since it does not detect the usage of types in comments or as strings

File: zimports/zimports.py (hunk 1 above)
Comment (CaselIT) [reply zimports/zimports.py]: will fix

**Comment by CaselIT** [comment]
@zzzeek I assume it's fine also the release workflow. this a copy of the stubs one

**Pull request merged**